Command/CreateDatabaseDoctrineCommand.php

```diff
@@ -60,51 +56,17 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             $params = $params['primary'];
         }
 
-        // Cannot inject `shard` option in parent::getDoctrineConnection
-        // cause it will try to connect to a non-existing database
-        if (isset($params['shards'])) {
-            $shards = $params['shards'];
-            // Default select global
-            $params = array_merge($params, $params['global'] ?? []);
-            unset($params['global']['dbname'], $params['global']['path'], $params['global']['url']);
-            if ($input->getOption('shard')) {
-                trigger_deprecation(
-                    'doctrine/doctrine-bundle',
-                    '2.7',
-                    'Passing a "shard" option for "%s" is deprecated. DBAL 3 does not support shards anymore.',
-                    self::class
-                );
-
-                foreach ($shards as $i => $shard) {
-                    if ($shard['id'] === (int) $input->getOption('shard')) {
-                        // Select sharded database
-                        $params = array_merge($params, $shard);
-                        unset($params['shards'][$i]['dbname'], $params['shards'][$i]['path'], $params['shards'][$i]['url'], $params['id']);
-                        break;
-                    }
-                }
-            }
-        }
-
         $hasPath = isset($params['path']);
         $name    = $hasPath ? $params['path'] : ($params['dbname'] ?? false);
         if (! $name) {
             throw new InvalidArgumentException("Connection does not contain a 'path' or 'dbname' parameter and cannot be created.");
         }
 
         // Need to get rid of _every_ occurrence of dbname from connection configuration and we have already extracted all relevant info from url
+        /** @psalm-suppress InvalidArrayOffset */
```

Comment:
Yes exactly. The error was about the `url` key.

> $connection->getParams() cannot really return url key, can it?

Actually it can :cry: I tested it. If I configure a DBAL connection using `url` in the bundle config then that key is also present in connection params.